flareon-cli/src/migration_generator.rs

```diff
@@ -428,6 +534,288 @@ impl AppState {
+enum VisibleSymbolKind {
+    Use,
+    Struct,
+    Const,
+}
+
+impl VisibleSymbol {
+    #[must_use]
+    fn new(alias: &str, full_path: &str, kind: VisibleSymbolKind) -> Self {
+        Self {
+            alias: alias.to_string(),
+            full_path: full_path.to_string(),
+            kind,
+        }
+    }
+
+    fn full_path_parts(&self) -> impl Iterator<Item = &str> {
+        self.full_path.split("::")
+    }
+
+    fn new_use(alias: &str, full_path: &str) -> Self {
+        Self::new(alias, full_path, VisibleSymbolKind::Use)
+    }
+
+    fn from_item_use(item: &syn::ItemUse, module_path: &ModulePath) -> Vec<Self> {
+        Self::from_tree(&item.tree, module_path)
+    }
+
+    fn from_item_struct(item: &syn::ItemStruct, module_path: &ModulePath) -> Self {
+        let ident = item.ident.to_string();
+        let full_path = Self::module_path(module_path, &ident);
+
+        Self {
+            alias: ident,
+            full_path,
+            kind: VisibleSymbolKind::Struct,
+        }
+    }
+
+    fn from_item_const(item: &syn::ItemConst, module_path: &ModulePath) -> Self {
+        let ident = item.ident.to_string();
+        let full_path = Self::module_path(module_path, &ident);
+
+        Self {
+            alias: ident,
+            full_path,
+            kind: VisibleSymbolKind::Const,
+        }
+    }
+
+    fn module_path(module_path: &ModulePath, ident: &str) -> String {
+        format!("{module_path}::{ident}")
+    }
+
+    fn from_tree(tree: &UseTree, current_module: &ModulePath) -> Vec<Self> {
+        match tree {
+            UseTree::Path(path) => {
+                let ident = path.ident.to_string();
+                let resolved_path = if ident == "crate" {
+                    current_module.crate_name().to_string()
+                } else if ident == "self" {
+                    current_module.to_string()
+                } else if ident == "super" {
+                    current_module.parent().to_string()
+                } else {
+                    ident
+                };
+
+                return Self::from_tree(&path.tree, current_module)
+                    .into_iter()
+                    .map(|import| {
+                        Self::new_use(
+                            &import.alias,
+                            &format!("{}::{}", resolved_path, import.full_path),
+                        )
+                    })
+                    .collect();
+            }
+            UseTree::Name(name) => {
+                let ident = name.ident.to_string();
+                return vec![Self::new_use(&ident, &ident)];
+            }
+            UseTree::Rename(rename) => {
+                return vec![Self::new_use(
+                    &rename.rename.to_string(),
+                    &rename.ident.to_string(),
+                )];
+            }
+            UseTree::Glob(_) => {
+                warn!("Glob imports are not supported");
+            }
```

Comment:
Can we not duplicate glob imports?

Reply:
Ah, this is one idea I haven't thought of. Well yes, that would work in most cases, but keep in mind we might have models from different source files in the same migration. This might _potentially_ cause import conflicts if we just copy the globs, but admittedly this would probably be a very rare case. (This is also the reason though why I opted for providing a full path in each use of each symbol instead of just copying `use` statements into the migration file)

Another sad thing about copying glob imports is that you need to copy all of them (and some will most likely end up being unused). There's no good solution unless we have a Rust compiler to resolve symbols for us, which is issue #67.

For now, I just created #69 since adding this now would complicate this overly complicated PR even more.

Reply:
Makes sense!